src/geo/transform.js

```diff
@@ -43,13 +43,17 @@ class Transform {
         if (this._minZoom === zoom) return;
         this._minZoom = zoom;
         this.zoom = Math.max(this.zoom, zoom);
+        if (this._maxZoom != null && this._minZoom > this._maxZoom)
+            throw new Error("maxZoom must always be greater than minZoom");
```

Comment:
Mapbox uses Error events to report user errors instead of throwing them. See:
https://github.com/mapbox/mapbox-gl-js/blob/master/CONTRIBUTING.md#code-conventions